packages/gatsby/src/utils/page-data.js

```diff
@@ -35,7 +38,88 @@ const write = async ({ publicDir }, page, result) => {
   await fs.outputFile(filePath, bodyStr)
 }
 
+const getChangedPageDataKeys = (state, cachedPageData) => {
+  if (cachedPageData && state.pageData) {
+    const pageKeys = []
+    state.pageData.forEach((newPageDataHash, key) => {
+      if (!cachedPageData.has(key)) {
+        pageKeys.push(key)
+      } else {
+        const previousPageDataHash = cachedPageData.get(key)
+        if (newPageDataHash !== previousPageDataHash) {
+          pageKeys.push(key)
+        }
+      }
+    })
+    return pageKeys
+  }
+
+  return [...state.pages.keys()]
+}
+
+const collectRemovedPageData = (state, cachedPageData) => {
+  if (cachedPageData && state.pageData) {
+    const deletedPageKeys = []
+    cachedPageData.forEach((_value, key) => {
+      if (!state.pageData.has(key)) {
+        deletedPageKeys.push(key)
+      }
+    })
+    return deletedPageKeys
+  }
+  return []
+}
+
+const checkAndRemoveEmptyDir = (publicDir, pagePath) => {
+  const pageHtmlDirectory = path.join(publicDir, pagePath)
+  const pageDataDirectory = path.join(
+    publicDir,
+    `page-data`,
+    fixedPagePath(pagePath)
+  )
+  const hasFiles = fs.readdirSync(pageHtmlDirectory)
+
+  // if page's html folder is empty also remove matching page-data folder
+  if (!hasFiles.length) {
+    fs.removeSync(pageHtmlDirectory)
+    fs.removeSync(pageDataDirectory)
+  }
+}
+
+const sortedPageKeysByNestedLevel = pageKeys =>
+  pageKeys.sort((a, b) => {
+    const currentPagePathValue = a.split(`/`).length
+    const previousPagePathValue = b.split(`/`).length
+    return currentPagePathValue > previousPagePathValue
+      ? -1
+      : currentPagePathValue < previousPagePathValue
+      ? 1
+      : 0
```

Comment:
this could entire block could be `return previousPagePathValue - currentPagePathValue`

Reply:
Done